src/sites/ptt/bot.ts

```diff
@@ -1,23 +1,29 @@
 import EventEmitter from 'eventemitter3';
 import sleep from 'sleep-promise';
 import Terminal from 'terminal.js';
-import decode from '../../utils/decode';
-import encode from '../../utils/encode';
 
-import key from '../../utils/keymap';
+import Socket from '../../socket';
+import {
+  decode,
+  encode,
+  keymap as key,
+} from '../../utils';
 import {
   getWidth,
   indexOfWidth,
   substrWidth,
 } from '../../utils/char';
+import Config from '../../config';
 
 import defaultConfig from './config';
+import {Article} from './Article';
+import {Board} from './Board';
 
-class Condition{
-  typeWord;
-  criteria;
+class Condition {
+  private typeWord: string;
+  private criteria: string;
 
-  constructor(type, criteria){
+  constructor(type: 'push'|'author'|'title', criteria: string) {
```

Comment:
```
enum Conditions {
    PUSH = 'push'
    ...
}

type ConditionType = keyof typeof Conditions;

constructor(type: ConditionType, ...) {
    ...
}
```

Reply:
I'll refactor this part in other PRs. Thanks.